### PR TITLE
Add additional information about CEs

### PR DESCRIPTION
**Pull request by gusty**
This will hopefully add some light about how to use properly lazy/strict CEs

### CODE REVIEW

File: docsrc/content/computation-expressions.fsx [hunk 1]
```diff
@@ -36,6 +36,21 @@ In other words:
  - ``monad.plus`` Lazy additive monadic builder. Use when you expect one or more results.
  - ``monad.plus'`` is the strict version of ``monad.plus``
 
+Note that a type is either lazy or strict, but it could act as fx or plus at the same time (see below some examples). This means that we need to pay attention when using a CE over a type, if the type is lazy but with use a strict monad, we'll get strict semantics which probably would make no sense, but if we do the opposite we might run into runtime errors, fortunatelly a compile-time warning (or error) will prevent us.
```
Comment (adz) [inline docsrc/content/computation-expressions.fsx:39]: typo on 'fortunatelly' -> fortunately

**Comment by adz** [comment]
I think it helps!

I'm thinking we should discuss how CE's work, and why it affects the monad builder. However, I haven't got my thoughts together well enough to do so...

**Comment by gusty** [comment]
I agree, the current CE documentation is not the best we can do.

File: docsrc/content/computation-expressions.fsx [hunk 2]
```diff
@@ -31,11 +31,26 @@ There is a single computation expression: ``monad`` but it comes in 4 flavours:
...
 
+Note that a type is either lazy or strict, but it could act as fx or plus at the same time (see below some examples). This means that we need to pay attention when using a CE over a type, if the type is lazy but with use a strict monad, we'll get strict semantics which probably would make no sense, but if we do the opposite we might run into runtime errors, fortunately a compile-time warning (or error) will prevent us.
+
+A simple way to find out if a type is strict or lazy is to execute this in fsi: `let _ : MyType<'t> = monad { printfn "I'm strict" }`
+
+For layered monads (monad transformers) the general rule is: the monad is strict unless at least one of its constituent types is lazy, in that case the whole monad becomes lazy.
+
+*)
+
+let _ : OptionT<list<unit option>> = monad { printfn "I'm strict" }
```
Comment (wallymathieu) [inline docsrc/content/computation-expressions.fsx:47]: I think that you need to import `FSharpPlus.Data`, while also include the reference to the F#+ dll (done in this file a bit further down)

**Review by wallymathieu — CHANGES_REQUESTED** [review]

**Comment by wallymathieu** [comment]
I think the text helps a bit to explain the difference. Nice to have a simple way to test it as well.

File: docsrc/content/computation-expressions.fsx (hunk 2 above)
Comment (gusty) [reply docsrc/content/computation-expressions.fsx:47]: Yes, maybe I can move the open declarations further up, so those one liners compile.

**Review by wallymathieu — APPROVED** [review]

**Pull request merged**